pkg/core/blockchain.go

```diff
@@ -358,13 +358,14 @@ func (bc *Blockchain) storeBlock(block *Block) error {
 		switch t := tx.Data.(type) {
 		case *transaction.RegisterTX:
 			assets[tx.Hash()] = &AssetState{
-				ID:        tx.Hash(),
-				AssetType: t.AssetType,
-				Name:      t.Name,
-				Amount:    t.Amount,
-				Precision: t.Precision,
-				Owner:     t.Owner,
-				Admin:     t.Admin,
+				ID:         tx.Hash(),
+				AssetType:  t.AssetType,
+				Name:       t.Name,
+				Amount:     t.Amount,
+				Precision:  t.Precision,
+				Owner:      t.Owner,
+				Admin:      t.Admin,
+				Expiration: bc.BlockHeight() + 1 + 2000000,
```

Comment:
hmmm
`+1 + 2000000` couldn't be constants with name (where name can explain why is that so)?

Reply:
It's so magic, that it actually is wrong, because for `RegisterTX` it's done like this:
```c#
#pragma warning disable CS0612                                                                                                                                                                                                                                                                                         
                        case RegisterTransaction tx_register:
                            snapshot.Assets.Add(tx.Hash, new AssetState
                            {
                                AssetId = tx_register.Hash,
                                AssetType = tx_register.AssetType,
                                Name = tx_register.Name,
                                Amount = tx_register.Amount,
                                Available = Fixed8.Zero,
                                Precision = tx_register.Precision,
                                Fee = Fixed8.Zero,
                                FeeAddress = new UInt160(),
                                Owner = tx_register.Owner,
                                Admin = tx_register.Admin,
                                Issuer = tx_register.Admin,
                                Expiration = block.Index + 2 * 2000000,
                                IsFrozen = false
                            });
                            break;
```
And it shouldn't be confused with `Neo.Asset.Create` interop that does it like this:
```c#
            AssetState asset = Snapshot.Assets.GetOrAdd(tx.Hash, () => new AssetState
            {
                AssetId = tx.Hash,
                AssetType = asset_type,
                Name = name,
                Amount = amount,
                Available = Fixed8.Zero,
                Precision = precision,
                Fee = Fixed8.Zero,
                FeeAddress = new UInt160(),
                Owner = owner,
                Admin = admin,
                Issuer = issuer,
                Expiration = Snapshot.Height + 1 + 2000000,
                IsFrozen = false
            });
```

Reply:
pow wow heh. It's good to document magic somewhere